src/game/objects/traps/lava.c

```diff
@@ -126,7 +127,9 @@ void LavaEmitter_Setup(OBJECT_INFO *obj)
     obj->control = LavaEmitter_Control;
     obj->draw_routine = Object_DrawDummyItem;
     obj->collision = Object_Collision;
-    obj->save_flags = 1;
+    if (g_Config.enable_enhanced_saves) {
+        obj->save_flags = 1;
+    }
```

Comment:
That's not good for the legacy files because now whether they'll work or not will be controlled be a JSON setting.
The code should be amended to always disregard save_flags for these objects, similar to how we have some hardcoded checks for the Bacon Lara. It's OK to set save_flags to 1 here unconditionally.

Reply:
By disregarding save flags, what exactly do you mean?

In `Savegame_Legacy_SaveToFile`, can I change `if (obj->save_flags)` to `if (obj->save_flags && item->object_number != O_LAVA_EMITTER && item->object_number != O_FLAME_EMITTER && item->object_number != O_WATERFALL) {`?

Reply:
Yes, that's exactly it, but it needs testing too. Basically make a save in original TombATI in The Great Pyramid and see if loading it in T1M with the proposed changes goes through or crashes the game.

Reply:
Gonna pose a multi part question sorry. Savegames are complicated to mess with and I want to make sure I don't break anything. : /

1. I made a Great Pyramid save near a lava emitter in TombATI. I tried to load it into T1M's 2.10.2 release and the game crashed. I added the `(obj->save_flags && item->object_number !=` check for `O_LAVA_EMITTER`, `O_FLAME_EMITTER`, and `O_WATERFALL` in `Savegame_Legacy_LoadFromFile`. That fixed the loading crash and the items behaved like TombATI where they do not restore on load. I also added those checks to `Savegame_Legacy_SaveToFile` in case someone ever saved legacy so those items that now have items flags do not save that way in legacy format. Does this sound right?

2. In `Savegame_Legacy_NeedsBaconLaraFix`, there's an `if (obj->save_flags)` check in that function, so I probably also need to add the ` item->object_number !=` checks there too for the emitters?

3. And lastly, for BSON saves, do I need to change `Savegame_BSON_LoadItems`'s `if (obj->save_flags) {` check similarly to the legacy version where it ignores certain objects unless the save is a newer version?

Reply:
1. Yes
2. Yes
3. I don't think you need to repeat this hardcoding of object ids, just make sure to only load the flags if the relevant bson keys were present in the document.

Reply:
I think all three of these are solved. The only one I'm not 100% sure on is `3`. But, I created a save with and without save_flags for the new objects. Then I logged the item loading for all save_flags and nothing in the load process seemed to change. I'm not sure how exactly but it looks to already be doing what you suggested.